javascripts/discourse/templates/components/custom-navigation.hbs

```diff
@@ -1,48 +1,52 @@
 {{#if shouldShow}}
   <div class="category-sidebar">
     <ul class="category-sidebar-list">
-        <li class="category-sidebar-list-item">
+        <li class="category-sidebar-list-item all-topics">
           {{#link-to "discovery.latest" class="category-sidebar-list-item-link"}}
             All Topics
           {{/link-to}}
         </li>
       {{#each sidebarCategories as |sidebarCategory|}}
-        <li class="category-sidebar-list-item">
+        <li 
+          class={{concat 'category-sidebar-list-item ' 
+          (if
+            (or 
+              (and
+                (eq currentRouteCategory sidebarCategory.name)
+                sidebarCategory.has_children
+              )
+              (childIsActive currentRoute sidebarCategory.slug)
+            ) 'show-children'
+          )}} 
+          style="--category-color: #{{sidebarCategory.color}}"
```

Comment:
Dynamic inline styles will cause both linter warnings and runtime warnings. But we can tackle that change after I add the CI setup.